src/main/java/io/conduit/SourceService.java

```diff
@@ -86,31 +88,44 @@ public void start(Source.Start.Request request, StreamObserver<Source.Start.Resp
         } catch (Exception e) {
             Logger.get().error("Error while starting.", e);
             responseObserver.onError(
-                    Status.INTERNAL.withDescription("couldn't start task: " + e.getMessage()).withCause(e).asException()
+                Status.INTERNAL.withDescription("couldn't start task: " + e.getMessage())
+                    .withCause(e)
+                    .asException()
             );
         }
     }
 
     @Override
     public StreamObserver<Source.Run.Request> run(StreamObserver<Source.Run.Response> responseObserver) {
         this.runStream = new DefaultSourceStream(
-                task,
-                position,
-                responseObserver,
-                Transformations::fromKafkaSource
+            task,
+            position,
+            responseObserver,
+            getTransformation()
         );
         runStream.startAsync();
         return runStream;
     }
 
+    private Function<SourceRecord, Record.Builder> getTransformation() {
+        if (cdcSupported()) {
+            return new DebeziumToOpenCDC();
+        }
+        return new KafkaToOpenCDC();
+    }
+
+    private boolean cdcSupported() {
+        return task.getClass().getCanonicalName().startsWith("io.debezium.connector");
+    }
```

Comment:
Looks to me like `cdcSupported` checks if we are wrapping a debezium connector (or do non-debezium connectors start with `io.debezium.connector`?). If so, I propose to rename the function to something with "debezium" in it.

Reply:
Yeah, initially I thought I'll end up with something generic, but I didn't. That's a good suggestion, I'll change.